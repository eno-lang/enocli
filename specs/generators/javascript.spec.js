const path = require('path');

const analyze = require('../../lib/analysis/analyze.js');
const generateJavascript = require('../../lib/generators/javascript.js');

const DOCUMENT_PATH = path.join(__dirname, '../fixtures/document.eno');
const TYPES_DOCUMENT_PATH = path.join(__dirname, '../fixtures/types.eno');

describe('generateJavascript', () => {
  it('generates javascript code from an abstract syntax tree', () => {
    const astDocument = analyze(DOCUMENT_PATH);
    const javascript = generateJavascript(astDocument);

    expect(javascript).toMatchSnapshot();
  });

  it('correctly exports inferred types', () => {
    const astDocument = analyze(TYPES_DOCUMENT_PATH);
    const javascript = generateJavascript(astDocument);

    expect(javascript).toMatchSnapshot();
  });
});
