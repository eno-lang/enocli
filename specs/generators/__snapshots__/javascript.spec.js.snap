// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateJavascript correctly exports inferred types 1`] = `
"const enolib = require('enolib');
const enotype = require('enotype');
const fs = require('fs');

enolib.register(enotype);

exports.parse = path => {
  const input = fs.readFileSync(path, 'utf-8');
  const _document = enolib.parse(input, { source: path });

  const document = {};
  document.boolean = _document.requiredField('boolean').requiredBooleanValue();
  document.color = _document.requiredField('color').requiredColorValue();
  document.date = _document.requiredField('date').requiredDateValue();
  document.datetime = _document.requiredField('datetime').requiredDatetimeValue();
  document.email = _document.requiredField('email').requiredEmailValue();
  document.float = _document.requiredField('float').requiredFloatValue();
  document.integer = _document.requiredField('integer').requiredIntegerValue();
  document.ipv4 = _document.requiredField('ipv4').requiredIpv4Value();
  document.latLng = _document.requiredField('lat_lng').requiredLatLngValue();
  document.json = _document.requiredField('json').requiredJsonValue();
  document.url = _document.requiredField('url').requiredUrlValue();
  document.floatList = _document.requiredList('float_list').requiredFloatValues();
  document.datetimeList = _document.requiredList('datetime_list').requiredDatetimeValues();
  document.jsonList = _document.requiredList('json_list').requiredJsonValues();

  return document;
};"
`;

exports[`generateJavascript generates javascript code from an abstract syntax tree 1`] = `
"const enolib = require('enolib');
const enotype = require('enotype');
const fs = require('fs');

enolib.register(enotype);

exports.parse = path => {
  const input = fs.readFileSync(path, 'utf-8');
  const _document = enolib.parse(input, { source: path });

  const document = {};
  document.empty = _document.requiredEmpty('empty') !== null;
  document.field = _document.requiredField('field').requiredStringValue();
  {
    const _fieldset = _document.requiredFieldset('fieldset');
    document.fieldset = {};
    document.fieldset.entry = _fieldset.requiredEntry('entry').requiredStringValue();
  }
  document.list = _document.requiredList('list').requiredStringValues();
  {
    document.section = {};
    const _section = _document.requiredSection('section');
    const section = document.section;
    section.field = _section.requiredField('field').requiredStringValue();
  }

  return document;
};"
`;
